### PR TITLE
fix block in close()

### PR DESCRIPTION
**Pull request by richard78917**
In case that close() is called while doing
read operation, the close() call is blocked and whole application
is stucked.

### CODE REVIEW

File: pynats/client.py [hunk 1]
```diff
@@ -279,7 +280,11 @@ def _readline(self, *, size: int = None) -> bytes:
         read = io.BytesIO()
 
         while True:
-            line = cast(bytes, self._socket_file.readline())
+            raw_bytes = self._socket_file.readline()
+            if not raw_bytes:
+                return None
```
Comment (Gr1N) [inline pynats/client.py]: The current signature of `_readline` function does not allow to return `None`. If you want to make such changes, then you need to update return type to `Optional[bytes]` and check all dependent code.

But I think such changes are out of scope.

**Comment by Gr1N** [comment]
Hi and thank you for your contribution.

Please fix a few moments:
1. Rebase your branch from master, I've fixed CI to run builds from PRs.
2. Use `[fix]` as a prefix for your commit message.
3. Update `CHANGELOG.md` using template: your_text, #11 by @richard78917

File: pynats/client.py (hunk 1 above)
Comment (richard78917) [reply pynats/client.py]: Ah, yes i missed that, thanks!
Maybe cleaner solution would be to raise an Exception here instead?

File: pynats/client.py [hunk 2]
```diff
@@ -279,7 +280,11 @@ def _readline(self, *, size: int = None) -> bytes:
         read = io.BytesIO()
 
         while True:
-            line = cast(bytes, self._socket_file.readline())
+            raw_bytes = self._socket_file.readline()
+            if not raw_bytes:
+                raise RuntimeError("unable to read from socket")
```
Comment (Gr1N) [inline pynats/client.py]: Exception is much better. But please introduce a new exception [here](https://github.com/Gr1N/nats-python/blob/master/pynats/exceptions.py) for your case and use it.

Also, it would be great to cover that case with tests.

Comment (richard78917) [reply pynats/client.py]: Updated as requested, thanks!

**Comment by Gr1N** [comment]
LGTM! Thank you for your contribution. I will release a new version of package this weekend.

**Comment by Gr1N** [comment]
New version of package just published:
- https://github.com/Gr1N/nats-python/releases/tag/0.7.0
- https://pypi.org/project/nats-python/0.7.0/

@richard78917 enjoy!
